apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.cs

```diff
@@ -239,6 +243,59 @@ internal UrlSigner WithSigningVersion(SigningVersion signingVersion)
                 request?.Headers?.ToDictionary(h => h.Key, h => h.Value),
                 request?.Content?.Headers?.ToDictionary(h => h.Key, h => h.Value));
 
+        /// <summary>
+        /// Creates a signed URL which can be used to provide limited access to specific buckets and objects to anyone
+        /// in possession of the URL, regardless of whether they have a Google account.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// When a <see cref="UrlSigner"/> is created with a service account credential, the signing can be performed
+        /// with no network access. When it is created with an implementation of <see cref="IBlobSigner"/>, that may require
+        /// network access or other IO. In that case, one of the asynchronous methods should be used when the caller is
+        /// in a context that should not block.
+        /// </para>
+        /// <para>
+        /// When the <paramref name="request"/> is specified, some of its headers will be included in the signed URL's
+        /// signature, and therefore must be included in requests made with the created URL. These are the Content-MD5 and
+        /// Content-Type content headers as well as any content or request header with a name starting with "x-goog-".
+        /// </para>
+        /// <para>
+        /// If <paramref name="request"/> is null, no headers are included in the signed URL's signature, so any requests
+        /// made with the created URL must not contain Content-MD5, Content-Type, or any header starting with "x-goog-".
```

Comment:
Is this w.r.t V2 or V4? IIUC you should still be able to pass in Content-Type in a V4 request if it's not part of the string_to_sign given signed headers.

Reply:
This is the general documentation - but it's copied and pasted from other overloads. I'm going to propose that when we start supporting V4, we move it all to a documentation page that the remarks can reference.

Reply:
Sg, thanks.